src/Task.cc

```diff
@@ -908,6 +927,30 @@ void Task::resume_execution(ResumeRequest how, WaitRequest wait_how,
   how_last_execution_resumed = how;
   set_debug_status(0);
 
+  last_resume_orig_cx = 0;
```

Comment:
This function is getting a bit gnarly. Can we move this code out into a couple of helper functions?